netty/src/test/java/io/grpc/netty/WriteQueueTest.java

```diff
@@ -61,14 +62,16 @@
 @RunWith(JUnit4.class)
 public class WriteQueueTest {
 
+  private final Object lock = new Object();
+
   @Mock
   public Channel channel;
 
   @Mock
   public ChannelPromise promise;
 
-  private long flushCalledNanos;
   private long writeCalledNanos;
+  private long flushCalledNanos = writeCalledNanos + 1;
```

Comment:
The write queue is to combine flushes. If it weren't for the flush combining it wouldn't serve a purpose. Continuing to use flush combining for window updates and similar seems a good idea, so I don't think we should flush directly.